From 3a54a590f1d3d70d886827f7c12104f89efa2e0b Mon Sep 17 00:00:00 2001
From: Andrej Lajovic <andrej.lajovic@ad-vega.si>
Date: Fri, 1 Aug 2014 15:58:45 +0200
Subject: [PATCH] Added the function wiringPiSPISetupWithMode()

This function is similar to wiringPiSPISetup() but accepts an additional
argument, namely the SPI mode.
---
 wiringPi/wiringPiSPI.c | 18 ++++++++++++++----
 wiringPi/wiringPiSPI.h |  4 ++++
 2 files changed, 18 insertions(+), 4 deletions(-)

diff --git a/wiringPi/wiringPiSPI.c b/wiringPi/wiringPiSPI.c
index 215a027..9a99449 100644
--- a/wiringPi/wiringPiSPI.c
+++ b/wiringPi/wiringPiSPI.c
@@ -28,7 +28,6 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/ioctl.h>
-#include <linux/spi/spidev.h>
 
 #include "wiringPi.h"
 
@@ -40,7 +39,6 @@
 
 const static char       *spiDev0  = "/dev/spidev0.0" ;
 const static char       *spiDev1  = "/dev/spidev0.1" ;
-const static uint8_t     spiMode  = 0 ;
 const static uint8_t     spiBPW   = 8 ;
 const static uint16_t    spiDelay = 0 ;
 
@@ -87,12 +85,12 @@ int wiringPiSPIDataRW (int channel, unsigned char *data, int len)
 
 
 /*
- * wiringPiSPISetup:
+ * wiringPiSPISetupWithMode:
  *	Open the SPI device, and set it up, etc.
  *********************************************************************************
  */
 
-int wiringPiSPISetup (int channel, int speed)
+int wiringPiSPISetupWithMode  (int channel, int speed, uint8_t spiMode)
 {
   int fd ;
 
@@ -119,3 +117,15 @@ int wiringPiSPISetup (int channel, int speed)
 
   return fd ;
 }
+
+
+/*
+ * wiringPiSPISetup:
+ *	Same as wiringPiSPISetupWithMode, but with implicit spiMode = SPI_MODE_0
+ *********************************************************************************
+ */
+
+int wiringPiSPISetup (int channel, int speed)
+{
+  return wiringPiSPISetupWithMode (channel, speed, SPI_MODE_0);
+}
diff --git a/wiringPi/wiringPiSPI.h b/wiringPi/wiringPiSPI.h
index f53697d..7492737 100644
--- a/wiringPi/wiringPiSPI.h
+++ b/wiringPi/wiringPiSPI.h
@@ -22,6 +22,9 @@
  ***********************************************************************
  */
 
+#include <stdint.h>
+#include <linux/spi/spidev.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -29,6 +32,7 @@ extern "C" {
 int wiringPiSPIGetFd  (int channel) ;
 int wiringPiSPIDataRW (int channel, unsigned char *data, int len) ;
 int wiringPiSPISetup  (int channel, int speed) ;
+int wiringPiSPISetupWithMode  (int channel, int speed, uint8_t spiMode) ;
 
 #ifdef __cplusplus
 }
-- 
2.1.0

